src/Ng/Catalog2Elfie.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Services.Metadata.Catalog;
+using NuGet.Services.Metadata.Catalog.Persistence;
+
+namespace Ng
+{
+    public class Catalog2Elfie
+    {
+        async Task Loop(Catalog2ElfieOptions options, CancellationToken cancellationToken)
+        {
+            Func<HttpMessageHandler> handlerFunc = CommandHelpers.GetHttpMessageHandlerFactory(options.Verbose, null, null);
+
+            Storage storage = options.StorageFactory.Create();
+            ReadWriteCursor front = new DurableCursor(storage.ResolveUri("cursor.json"), storage, MemoryCursor.Min.Value);
+            ReadCursor back = MemoryCursor.Max;
+
+            CommitCollector collector = new ElfieFromCatalogCollector(new Uri(options.Source), storage, options.MaxThreads, handlerFunc);
+
+            while (true)
+            {
+                // Keep running as long as there are more packages to process.
+                bool run = false;
+                do
+                {
+                    run = await collector.Run(front, back, cancellationToken);
+                }
+                while (run);
+
+                // When there are no more packages to process, either stop the application or 
+                // wait a few seconds before checking for new packages.
+                if (options.Interval <= 0)
+                {
+                    break;
+                }
+                else
+                {
+                    Thread.Sleep(options.Interval * 1000);
+                }
+            }
+        }
+
+        void PrintUsage()
+        {
+            Console.WriteLine();
+            Console.WriteLine("Usage: ng catalog2elfie -source <catalog> -storageBaseAddress <storage-base-address> -storageType file|azure [-storagePath <path>]|[-storageAccountName <azure-acc> -storageKeyValue <azure-key> -storageContainer <azure-container> -storagePath <path>] [-verbose true|false] [-interval <seconds>] [-maxthreads <int>]");
+            Console.WriteLine("Example: ng.exe catalog2elfie -source http://api.nuget.org/v3/catalog0/index.json -storageBaseAddress file:///C:/NuGet -storageType file -storagePath C:\\NuGet -verbose true");
```

Comment:
I'll update the usage statement to add descriptions for all the parameters. 

To answer your specific question, storagePath is only used when the storageType==file. It's the directory where the files will be saved. storageBaseAddress makes more sense when the storageType==azure. In this case, storageBaseAddress is the URL to your Azure blob. When storageType==file, storageBaseAddress is really only used to calculate relative paths. For example, if you tell the Storage type to save a stream to file:///C:/NuGet/Idx/Newtonsoft.Json/8.0.2/8.0.2.idx, it removes storageBaseAddress from the URL to determine the relative path (e.g. Idx/Newtonsoft.Json/8.0.2/8.0.2.idx.) Then it saves the file to {storagePath} + {relativePath} = C:\NuGet\Idx\Newtonsoft.Json/8.0.2/8.0.2.idx.